File: src/data/catalog.ts
import { Category } from '../core/types';

/**
 * Browse categories shown on the Search screen.
 *
 * These are not mock content: each one runs a real search against the
 * provider when tapped. The colours are icon-ink accents from the Nocturne
 * Cyan system — the tiles themselves are drawn in Cards.tsx's CATEGORY_ART.
 */
export const BROWSE_CATEGORIES: Category[] = [
  { id: 'c1', name: 'Charts', color: '#35D6C6', query: 'top hits this week' },
  { id: 'c2', name: 'New Releases', color: '#FFFFFF', query: 'new music releases' },
  { id: 'c3', name: 'Moods', color: '#FFA63D', query: 'chill mood playlist' },
  { id: 'c4', name: 'Indian', color: '#F5A9CE', query: 'bollywood hits' },
  { id: 'c5', name: 'Hip-Hop', color: '#FF8A3D', query: 'hip hop essentials' },
  { id: 'c6', name: 'Pop', color: '#FFFFFF', query: 'pop hits' },
  { id: 'c7', name: 'EDM', color: '#57E8D6', query: 'edm dance mix' },
  { id: 'c8', name: 'Rock', color: '#FF6B5A', query: 'rock classics' },
  { id: 'c9', name: 'Jazz', color: '#7FB2FF', query: 'jazz classics' },
  { id: 'c10', name: 'Classical', color: '#E4D7A8', query: 'classical instrumental' },
  { id: 'c11', name: 'Lo-Fi', color: '#9B8CF0', query: 'lofi beats to relax' },
  { id: 'c12', name: 'Workout', color: '#5AD1A8', query: 'workout motivation songs' },
  { id: 'c13', name: 'Party', color: '#FFC24D', query: 'party anthems' },
  { id: 'c14', name: 'Relax', color: '#7FE3D4', query: 'relaxing ambient music' },
];

/** How many tiles the Search grid shows before "See all" is tapped. */
export const BROWSE_VISIBLE_COUNT = 8;

/**
 * Query pools for the Home quick actions.
 *
 * Each tap picks one at random so the tile does not return the same list
 * every time. Liked is absent because it plays from the local library.
 */
export const ACTION_QUERIES: Record<string, string[]> = {
  discover: [
    'trending songs this week',
    'viral hits right now',
    'top global chart songs',
    'new music this month',
    'breakout artists 2026',
    'most played songs today',
  ],
  chill: [
    'chill relaxing songs',
    'lofi chill beats',
    'acoustic chill playlist',
    'calm indie chill',
    'soft rnb chill',
    'sunset chill mix',
  ],
  focus: [
    'focus instrumental concentration',
    'deep focus study music',
    'ambient focus no lyrics',
    'piano focus instrumental',
    'minimal techno focus',
    'nature focus soundscape',
  ],
};

/** A random query for a quick action, or null when it has no pool. */
export function randomQueryFor(actionId: string): string | null {
  const pool = ACTION_QUERIES[actionId];
  if (!pool || pool.length === 0) return null;
  return pool[Math.floor(Math.random() * pool.length)];
}

/** The quick-action tiles on Home, each backed by a real query. */
export const QUICK_ACTIONS = [
  { id: 'liked', label: 'Liked', query: null },
  { id: 'discover', label: 'Discover', query: 'discover new music' },
  { id: 'chill', label: 'Chill', query: 'chill relaxing songs' },
  { id: 'focus', label: 'Focus', query: 'focus instrumental concentration' },
] as const;

/** Seeds the Home featured card. */
export const FEATURED_QUERY = 'calm ambient evening';

File: app/src/main/java/com/example/arkify/data/Catalog.kt
package com.example.arkify.data

import com.example.arkify.model.Artist
import com.example.arkify.model.Category
import com.example.arkify.model.Track

object Catalog {
    val BROWSE_CATEGORIES = listOf(
        Category("c1", "Charts", "#35D6C6", "top hits this week"),
        Category("c2", "New Releases", "#FFFFFF", "new music releases"),
        Category("c3", "Moods", "#FFA63D", "chill mood playlist"),
        Category("c4", "Indian", "#F5A9CE", "bollywood hits"),
        Category("c5", "Hip-Hop", "#FF8A3D", "hip hop essentials"),
        Category("c6", "Pop", "#FFFFFF", "pop hits"),
        Category("c7", "EDM", "#57E8D6", "edm dance mix"),
        Category("c8", "Rock", "#FF6B5A", "rock classics"),
        Category("c9", "Jazz", "#7FB2FF", "jazz classics"),
        Category("c10", "Classical", "#E4D7A8", "classical instrumental"),
        Category("c11", "Lo-Fi", "#9B8CF0", "lofi beats to relax"),
        Category("c12", "Workout", "#5AD1A8", "workout motivation songs"),
        Category("c13", "Party", "#FFC24D", "party anthems"),
        Category("c14", "Relax", "#7FE3D4", "relaxing ambient music")
    )

    val QUICK_ACTIONS = listOf(
        QuickAction("liked", "Liked"),
        QuickAction("discover", "Discover"),
        QuickAction("chill", "Chill"),
        QuickAction("focus", "Focus")
    )

    data class QuickAction(val id: String, val label: String)

    // Curated high-fidelity playable catalog with royalty-free & direct stream audio
    val CURATED_FEATURED_TRACKS = listOf(
        Track(
            id = "arkify:ambient-flow-01",
            title = "Midnight Echoes",
            artist = Artist("art-1", "Aura Nocturne"),
            albumImageUrl = "https://images.unsplash.com/photo-1518709268805-4e9042af9f23?w=600&auto=format&fit=crop&q=80",
            duration = 195,
            audioUrl = "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3",
            sourceId = "featured_1",
            album = "Cinematic Horizons"
        ),
        Track(
            id = "arkify:chill-beats-02",
            title = "Cyan Horizon",
            artist = Artist("art-2", "Solaris Drift"),
            albumImageUrl = "https://images.unsplash.com/photo-1508700115892-45ecd05ae2ad?w=600&auto=format&fit=crop&q=80",
            duration = 240,
            audioUrl = "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-2.mp3",
            sourceId = "featured_2",
            album = "Nocturne Dreamscapes"
        ),
        Track(
            id = "arkify:lofi-focus-03",
            title = "Rain Over Neon",
            artist = Artist("art-3", "Komorebi"),
            albumImageUrl = "https://images.unsplash.com/photo-1511671782779-c97d3d27a1d4?w=600&auto=format&fit=crop&q=80",
            duration = 184,
            audioUrl = "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-3.mp3",
            sourceId = "featured_3",
            album = "Lo-Fi Study Session"
        ),
        Track(
            id = "arkify:deep-focus-04",
            title = "Quantum Solitude",
            artist = Artist("art-4", "Starlight Audio"),
            albumImageUrl = "https://images.unsplash.com/photo-1470225620780-dba8ba36b745?w=600&auto=format&fit=crop&q=80",
            duration = 210,
            audioUrl = "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-4.mp3",
            sourceId = "featured_4",
            album = "Deep Space Pulse"
        ),
        Track(
            id = "arkify:electronic-vibe-05",
            title = "Velvet Skyline",
            artist = Artist("art-5", "Synthwave Syndicate"),
            albumImageUrl = "https://images.unsplash.com/photo-1493225457124-a3eb161ffa5f?w=600&auto=format&fit=crop&q=80",
            duration = 222,
            audioUrl = "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-5.mp3",
            sourceId = "featured_5",
            album = "Retro Glow"
        ),
        Track(
            id = "arkify:peaceful-piano-06",
            title = "Subtle Reflections",
            artist = Artist("art-6", "Elysian"),
            albumImageUrl = "https://images.unsplash.com/photo-1520523839898-507121c25567?w=600&auto=format&fit=crop&q=80",
            duration = 178,
            audioUrl = "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-6.mp3",
            sourceId = "featured_6",
            album = "Acoustic Whispers"
        )
    )
}
